src/V3EmitCSyms.cpp

```diff
@@ -981,52 +811,216 @@ void EmitCSyms::emitSymImp() {
+            stmt += ", ";
+            stmt += std::to_string(pdim);
+            stmt += bounds;
+            stmt += ");";
+            add(stmt);
         }
+    }
 
-        closeSplit();
-        puts("}\n");
+    return stmts;
+}
+
+std::vector<std::string> EmitCSyms::getSymDtorStmts() {
+    std::vector<std::string> stmts;
+
+    const auto add = [&stmts](const std::string& stmt) { stmts.emplace_back(stmt); };
+
+    emitScopeHier(stmts, true);
+    if (v3Global.needTraceDumper()) add("if (__Vm_dumping) _traceDumpClose();");
+    if (v3Global.opt.profPgo()) {
+        add("_vm_pgoProfiler.write(\"" + topClassName()
+            + "\", _vm_contextp__->profVltFilename());");
+    }
+    add("// Tear down sub module instances");
+    for (const ScopeModPair& itpair : vlstd::reverse_view(m_scopes)) {
+        const AstScope* const scopep = itpair.first;
+        const AstNodeModule* const modp = itpair.second;
+        if (modp->isTop()) continue;
+        add(protect(scopep->nameDotless()) + ".dtor();");
+    }
+    return stmts;
+}
+
+void EmitCSyms::emitSplit(std::vector<std::string>& stmts, const std::string name,
+                          size_t maxStmts) {
+    const std::string baseName = m_symsFileBase + "__" + name;
+    size_t nSplits = 0;
+    while (stmts.size() > 1) {
+        size_t nSubFunctions = 0;
+        for (size_t i = 0; i < stmts.size(); i += maxStmts) {
+            const std::string nStr = std::to_string(nSplits++);
+            // Name of sub-function we are emitting now
+            const std::string funcName = symClassName() + "__" + name + "__" + nStr;
+            m_splitFuncNames.emplace_back(funcName);
+
+            // Open split file
+            openOutputFile(baseName + "__" + nStr + "__Slow.cpp");
+            // Emit header
+            emitSymImpPreamble();
+            // Open sub-function definition in the split file
+            puts("void " + symClassName() + "::" + funcName + "() {\n");
+            // Emit statements
+            for (size_t j = 0; j < maxStmts; ++j) {
+                if (i + j >= stmts.size()) break;
+                m_ofp->putsNoTracking("    ");
+                m_ofp->putsNoTracking(stmts.at(i + j));
+                m_ofp->putsNoTracking("\n");
+            }
+            // Close sub-function
+            puts("}\n");
+            // Close split file
+            closeOutputFile();
+
+            // Replace statements with a call to the sub-function
+            stmts[nSubFunctions++] = funcName + "();";
+        }
+        stmts.resize(nSubFunctions);
+    }
+}
+
+void EmitCSyms::emitSymImp(AstNetlist* netlistp) {
+    UINFO(6, __FUNCTION__ << ": ");
+
+    const size_t optSplit = static_cast<size_t>(v3Global.opt.outputSplitCFuncs());
+    // Allow at least 2 statements per function, to ensure tree reduction
+    // actually makes forward progress. Also assume each statement is worth 5
+    // units as each of these are somewhat complicated, not just simple ops.
```

Comment:
Please form this using some math of INSTR_COUNT_'s.

Reply:
Not sure what you would like here, can you please elaborate?